src/KineticConvolution/Direction.cs

```diff
@@ -31,12 +31,132 @@ public Direction(IAlgebraicNumber x, IAlgebraicNumber y)
 
         public IAlgebraicNumber Y { get; }
 
-        public bool Equals(Direction d)
+        /// <inheritdoc />
+        public bool Equals(Direction? other)
         {
-            return
-                DirectionHelper.Determinant(this, d).IsZero() &&
-                X.Sign() == d.X.Sign() &&
-                Y.Sign() == d.Y.Sign();
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return DirectionHelper.Determinant(this, other).IsZero()
+                && X.Sign() == other.X.Sign()
+                && Y.Sign() == other.Y.Sign();
+        }
+
+        public int Compare(Direction direction1, Direction direction2)
+        {
+            if (direction1.Equals(direction2))
+            {
+                return 0;
+            }
+
+            return direction1.BelongsTo(
+                       new DirectionRange(
+                           this,
+                           direction2,
+                           Orientation.CounterClockwise))
+                       ? 1
+                       : -1;
+        }
```

Comment:
We can't, `IComparable` asks for a `CompareTo` method that takes one other direction. In that case, we compare two directions which is compared with the reference "this". Maybe the name Compare can be improved though. @thomashilke  ?

Reply:
We could call it `IsFirstCounterClockwise`, since it returns 1 if `direction1` is encountered before `direction2` when turning counterclockwise and starting from the current direction.

Reply:
renamed in CompareTo and FirstOf / LastOf